byway/src/main/webapp/home.js

```diff
@@ -86,22 +87,46 @@ function showIncompleteTrip(tripNum, trip, isDestinationsMissing, isInterestsMis
  * @param {Number} tripNum 
  * @param {Trip} trip
  */
-function showCompleteTrip(tripNum, trip){
+async function showCompleteTrip(trip){
   let container = document.getElementById("past-trips-container");
   let pastTrip = document.createElement('div');
   pastTrip.className = "past-trip";
   let title = document.createElement('a');
+  title.id = "title-" + trip.keyString;
   pastTrip.append(title);
   let mapContainer = document.createElement('div');
   mapContainer.className = 'map';
-  mapContainer.id = "map-" + trip.keyString
+  mapContainer.id = "map-" + trip.keyString;
   pastTrip.append(mapContainer);
   container.append(pastTrip);
   initMap(trip.start, trip.start, trip.route, trip.keyString);
-  title.innerText = "Trip #" + tripNum;
+  let titleString = await constructTripTitle(trip);
+  title.innerText = titleString;
   title.href = configureTripKeyForPath(trip.keyString, "/routepage.html");
 }
 
+async function constructTripTitle(trip){
+  //let title = document.getElementById('title-' + keyString);
```

Comment:
Make sure you remove these comments before submitting this PR

Reply:
done